packages/curve-ui-kit/src/shared/ui/Balance.tsx

```diff
@@ -0,0 +1,106 @@
+import AccountBalanceWalletOutlinedIcon from '@mui/icons-material/AccountBalanceWalletOutlined'
+import Button from '@mui/material/Button'
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { Spacing, IconSize } = SizesAndSpaces
+
+const formatNumber = (value: number): string =>
+  value.toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+
+type MaxButtonProps = {
+  children: React.ReactNode
+  underline: boolean
+  onClick?: () => void
+}
+
+/** Reusable Max button component with consistent styling */
+const MaxButton = ({ children, underline, onClick }: MaxButtonProps) => (
+  <Button
+    color="ghost"
+    variant="link"
+    size="extraSmall"
+    onClick={onClick}
+    sx={{
+      minWidth: 'unset',
+      // Sadly the !important is still necessary because of the !important in defineMuiButton for the Link variant :(
+      '&': { height: '0px !important' },
+      ...(underline && {
+        '&:hover .balance': {
+          textDecoration: 'underline',
+        },
+      }),
```

Comment:
This shouldn't be a link variant just a ghost color - then your overrides shouldn't be needed
```suggestion
    size="extraSmall"
    onClick={onClick}
```